hls-test-utils/src/Test/Hls/FileSystem.hs

```diff
@@ -66,6 +67,7 @@ data VirtualFileTree =
 data FileTree
   = File FilePath Content
   | Directory FilePath [FileTree]
+  | CopiedDirectory FilePath
```

Comment:
how is this different? some haddock would be helpful

Reply:
Looks like it corresponds to a directory glob.

Reply:
I guess I'm confused about how it differs from the previous entry :thinking: Do we need two ways of specifying directories?

Reply:
I have to manually specified the whole virtual tree down to the file levels.
`CopiedDirectory` allow us to simply copied an existing folder to vfs.
Matching those test in ghcide depend on just a folder in the `test/data`.
Yep, maybe a doc comment need to be added

Reply:
added